internal/services/notification/runevents.go

```diff
@@ -97,24 +117,66 @@ func (n *NotificationService) runEventsHandler(ctx context.Context) error {
 				return errors.WithStack(err)
 			}
 
-			// TODO(sgotti)
-			// this is just a basic handling. Improve it to store received events and
-			// their status in the db so we can also do more logic like retrying and handle
-			// multiple kind of notifications (email etc...)
+			if ev.Sequence < expectedRunEventSequence {
+				return errors.Errorf("run event %d already processed", ev.Sequence)
```

Comment:
See the previous comment https://github.com/agola-io/agola/pull/423/files#r1297187693

"ignore sequences before the expected one"